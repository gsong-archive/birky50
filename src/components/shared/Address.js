//@flow strict
import React from "react";

import { Link } from "../../styles/components";
import { Location } from "./EmojiLabels";

type Props = {
  address: string,
  url: string,
};

export default ({ address, url }: Props) => (
  <div>
    <Location label={<Link href={url}>{address}</Link>} />
  </div>
);
