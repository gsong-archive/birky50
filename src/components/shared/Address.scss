.address-link {
  padding-left: 1.5rem;
  text-indent: -1.5rem;
}
