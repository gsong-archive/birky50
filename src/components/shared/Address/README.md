```js
<Address
  address="123 Main Street, Portland, OR 97217"
  url="https://example.com"
/>
```
