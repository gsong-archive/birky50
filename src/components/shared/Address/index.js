//@flow strict
import React from "react";

import { Link } from "../../../styles/components";
import { Location } from "../EmojiLabels";

type Props = {
  address: string,
  /** Link to map */
  url: string,
};

/**
 * Render an address set inside a link (preferably a map)
 */
export default ({ address, url }: Props) => (
  <div>
    <Location label={<Link href={url}>{address}</Link>} />
  </div>
);
