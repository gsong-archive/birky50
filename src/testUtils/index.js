import * as faker from "faker";

const address = () =>
  [
    faker.address.streetAddress(),
    faker.address.city(),
    `${faker.address.stateAbbr()} ${faker.address.zipCode()}`,
  ].join(", ");

const card = () => ({
  name: `${faker.name.firstName()} ${faker.name.lastName()}`,
  url: faker.internet.url(),
  price: faker.random.word(),
  address: address(),
  addressUrl: faker.internet.url(),
  phone: faker.phone.phoneNumber(),
  imgUrl: faker.image.imageUrl(),
});

const generate = {
  address,
  card,
  phoneNumber: faker.phone.phoneNumber,
  url: faker.internet.url,
  words: faker.random.words,
};

export { generate };
