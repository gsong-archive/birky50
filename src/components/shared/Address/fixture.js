//@flow
import { createFixture } from "react-cosmos-flow/fixture";

import { generate } from "testUtils";

import Address from "./index";

export default createFixture({
  component: Address,
  name: "Address",
  props: { address: generate.address(), url: generate.url() },
});
