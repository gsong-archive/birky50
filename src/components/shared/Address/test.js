//@flow strict
import React from "react";

import { render } from "react-testing-library";

import { generate } from "testUtils";

import Address from "./index";

test("User can see address and map link", () => {
  const props = { address: generate.address(), url: generate.url() };
  const { container } = render(<Address {...props} />);
  const link = container.querySelector("a");

  expect(container).toHaveTextContent(props.address);
  // $FlowFixMe
  expect(link.href).toMatch(props.url);
});

test("Address is rendered", () => {
  const props = {
    address: "123 Main Street",
    url: "https://example.com",
  };
  const { container } = render(<Address {...props} />);

  expect(container.firstChild).toMatchSnapshot();
});
