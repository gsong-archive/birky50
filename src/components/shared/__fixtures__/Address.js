import { generate } from "testUtils";

import Address from "../Address";

export default {
  component: Address,
  props: { address: generate.address(), url: generate.url() },
};
