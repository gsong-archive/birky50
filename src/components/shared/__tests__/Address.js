import React from "react";

import { render } from "react-testing-library";

import { generate } from "testUtils";

import Address from "../Address";

test("User can see address and map link", () => {
  const props = { address: generate.address(), url: generate.url() };
  const { container } = render(<Address {...props} />);
  const link = container.querySelector("a");

  expect(container.textContent).toMatch(props.address);
  expect(link.href).toMatch(props.url);
});

test("Address is rendered", () => {
  const props = {
    address: "123 Main Street",
    url: "https://example.com",
  };
  const { container } = render(<Address {...props} />);

  expect(container.firstChild).toMatchSnapshot();
});
