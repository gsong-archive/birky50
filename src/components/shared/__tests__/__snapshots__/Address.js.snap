// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address is rendered 1`] = `
.emotion-0 {
  padding-left: 1.5rem;
  text-indent: -1.5rem;
}

<div
  class="emotion-0"
>
  <span
    role="img"
    aria-label="round pushpin"
  >
    📍
  </span>
  <a
    href="https://example.com"
  >
    123 Main Street
  </a>
</div>
`;
